src/components/KuadrantTLSPage.tsx

```diff
@@ -0,0 +1,306 @@
+      setSelectedNamespace(parsedYaml.metadata?.namespace || '');
+      setSelectedGateway(parsedYaml.spec?.targetRef?.name || '');
+      setSelectedCertIssuer(parsedYaml.spec?.issuerRef?.name || '');
+    } catch (e) {
+      console.error('Error parsing YAML:', e);
+    }
+  };
+
+  React.useEffect(() => {
+    const updatedTLSState = createTlsPolicy();
+    setYamlInput((updatedTLSState));
+  }, [policyName, selectedNamespace, selectedGateway, selectedCertIssuer]);
+
+  const [view, setView] = React.useState('form');
+
+
+  // Policy Name
+  const handleNameChange = (_event, policyName: string) => {
+    setPolicyName(policyName);
+  };
+
+  // Gateway selector
+  const tlsPolicyGVK = getGroupVersionKindForResource({ apiVersion: 'kuadrant.io/v1alpha1', kind: 'TLSPolicy' });
+  const [tlsPolicyModel] = useK8sModel({ group: tlsPolicyGVK.group, version: tlsPolicyGVK.version, kind: tlsPolicyGVK.kind });
+
+  const gatewayResource = {
+    groupVersionKind: { group: 'gateway.networking.k8s.io', version: 'v1', kind: 'Gateway' },
+    isList: true
+  };
+
+  const [gatewayData, gatewayLoaded, gatewayError] = useK8sWatchResource(gatewayResource);
+  React.useEffect(() => {
+
+    if (gatewayLoaded && !gatewayError && Array.isArray(gatewayData)) {
+      setGateways(gatewayData.map((ns) => ns.metadata.name));
+    }
+  }, [gatewayData, gatewayLoaded, gatewayError]);
+
+  const handleGatewayChange = (event) => {
+    setSelectedGateway(event.currentTarget.value);
+  };
+
+  // Cert manager cert issuer selector
+  const certIssuerResource = {
+    groupVersionKind: { group: 'cert-manager.io', version: 'v1', kind: 'ClusterIssuer' },
+    isList: true
+  };
+  const [certIssuerData, certIssuerLoaded, certIssuerError] = useK8sWatchResource(certIssuerResource);
+  React.useEffect(() => {
+
+    if (certIssuerLoaded && !certIssuerError && Array.isArray(certIssuerData)) {
+      setCertIssuers(certIssuerData.map((ns) => ns.metadata.name));
+    }
+  }, [certIssuerData, certIssuerLoaded, certIssuerData]);
+
+  const handleCertIssuerChange = (event) => {
+    setSelectedCertIssuer(event.currentTarget.value);
+  };
+
+  // Create
+  const handleCreateResource = () => {
+    handleCreate(tlsPolicyModel, tlsPolicy, selectedNamespace, "TLSPolicy", history);
+  };
+
+
+  //Cancel
+  const handleCancelResource = () => {
+    handleCancel(selectedNamespace, "tlsPolicy", tlsPolicy, history);
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title data-test="example-page-title"> Create TLS Policy </title>
+      </Helmet>
+      <Page>
+        <PageSection className='pf-m-no-padding'>
+          <div className='co-m-nav-title'>
+
+            <Title headingLevel="h1">{('Create TLS Policy')}</Title>
+            <p className='help-block co-m-pane__heading-help-text'>
+              <div> Targets Gateway API networking resources Gateways to provide tls for gateway listeners by managing the lifecycle of tls certificates using CertManager</div>
```

Comment:
probably want to i18n this in the very excellently named `locales/en/plugin__console-plugin-template.json`